### PR TITLE
Use snapshots for component-stack tests

### PR DESCRIPTION
**Pull request by timneutkens**
### What?

Follow-up to #60579 and #60750. Checking `startsWith` is not enough because it hides the rest of the stack. Changed the test to check the snapshot for Turbopack and webpack.

Fixes a bug where the stack lines showed `http (NaN:NaN)` as the source lines.

Added support for source lines that don't have a open in editor.

Closes NEXT-2128

### CODE REVIEW

**Comment by timneutkens** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/vercel/next.js/60768?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

<b>In this stack:</b> <code>🔄 Update ReactRefreshRegression test snapshot, Use snapshots for component-stack tests</code><a href="https://withgraphite.notion.site/Graphite-beta-features-EXTERNAL-496843566b204c7a97a2a21f8dea5434">(?)</a>

* `canary`
  * **#60767** <a href="https://app.graphite.dev/github/pr/vercel/next.js/60767?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **#60768** <a href="https://app.graphite.dev/github/pr/vercel/next.js/60768?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @timneutkens and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: test/development/acceptance-app/component-stack.test.ts [hunk 1]
```diff
@@ -50,13 +50,72 @@ describe('Component Stack in error overlay', () => {
...
-      expected
-    )
+    if (process.env.TURBOPACK) {
+      expect(await session.getRedboxComponentStack()).toMatchInlineSnapshot(`
+        "p
+        div
+        Component
+        main
+        InnerLayoutRouter
+        http (NaN:NaN)
```
Comment (timneutkens) [inline test/development/acceptance-app/component-stack.test.ts:60]: @kwonoj these `http (NaN:NaN)` parts are unexpected

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 3299d04182f0ce9e00a2e8419fb74803f4c501a5

`pnpm test-dev test/e2e/app-dir/not-found/basic/index.test.ts` (PPR)
- app dir - not-found - basic > with runtime = edge > should render the 404 page when the file is removed, and restore the page when re-added

<details>
<summary>Expand output</summary>

  ● app dir - not-found - basic › with runtime = edge › should render the 404 page when the file is removed, and restore the page when re-added

    TIMED OUT: Root Not Found

    My page

    undefined

      636 |
      637 |   if (hardError) {
    > 638 |     throw new Error('TIMED OUT: ' + regex + '\n\n' + content + '\n\n' + lastErr)
          |           ^
      639 |   }
      640 |   return false
      641 | }

      at check (lib/next-test-utils.ts:638:11)
      at Object.<anonymous> (e2e/app-dir/not-found/basic/index.test.ts:123:11)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Comment by ijjk** [comment]
# Stats from current PR

<details open>
<summary><strong>Default Build</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 01-17-Use_snapshots_for_component-stack_tests | Change |
| - | - | - | - |
| buildDuration | 13.1s | 13.1s | N/A |
| buildDurationCached | 7.3s | 6.2s | N/A |
| nodeModulesSize | 200 MB | 200 MB | ✓ |
| nextStartRea..uration (ms) | 413ms | 411ms | N/A |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 01-17-Use_snapshots_for_component-stack_tests | Change |
| - | - | - | - |
| 193.HASH.js gzip | 181 B | 182 B | N/A |
| 3f784ff6-HASH.js gzip | 53.4 kB | 53.4 kB | ✓ |
| 433-HASH.js gzip | 28.9 kB | 28.9 kB | N/A |
| framework-HASH.js gzip | 45.2 kB | 45.2 kB | ✓ |
| main-app-HASH.js gzip | 239 B | 242 B | N/A |
| main-HASH.js gzip | 31.8 kB | 31.8 kB | N/A |
| webpack-HASH.js gzip | 1.7 kB | 1.7 kB | N/A |
| Overall change | 98.6 kB | 98.6 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 01-17-Use_snapshots_for_component-stack_tests | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 01-17-Use_snapshots_for_component-stack_tests | Change |
| - | - | - | - |
| _app-HASH.js gzip | 194 B | 195 B | N/A |
| _error-HASH.js gzip | 183 B | 181 B | N/A |
| amp-HASH.js gzip | 504 B | 502 B | N/A |
| css-HASH.js gzip | 321 B | 321 B | ✓ |
| dynamic-HASH.js gzip | 2.5 kB | 2.5 kB | N/A |
| edge-ssr-HASH.js gzip | 255 B | 253 B | N/A |
| head-HASH.js gzip | 350 B | 349 B | N/A |
| hooks-HASH.js gzip | 369 B | 369 B | ✓ |
| image-HASH.js gzip | 4.28 kB | 4.28 kB | N/A |
| index-HASH.js gzip | 255 B | 256 B | N/A |
| link-HASH.js gzip | 2.61 kB | 2.61 kB | ✓ |
| routerDirect..HASH.js gzip | 312 B | 311 B | N/A |
| script-HASH.js gzip | 385 B | 383 B | N/A |
| withRouter-HASH.js gzip | 307 B | 308 B | N/A |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 3.4 kB | 3.4 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 01-17-Use_snapshots_for_component-stack_tests | Change |
| - | - | - | - |
| _buildManifest.js gzip | 483 B | 484 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 01-17-Use_snapshots_for_component-stack_tests | Change |
| - | - | - | - |
| index.html gzip | 528 B | 527 B | N/A |
| link.html gzip | 540 B | 539 B | N/A |
| withRouter.html gzip | 524 B | 522 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 01-17-Use_snapshots_for_component-stack_tests | Change |
| - | - | - | - |
| edge-ssr.js gzip | 93.9 kB | 93.9 kB | N/A |
| page.js gzip | 148 kB | 148 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Middleware size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 01-17-Use_snapshots_for_component-stack_tests | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 624 B | 626 B | N/A |
| middleware-r..fest.js gzip | 151 B | 151 B | ✓ |
| middleware.js gzip | 37.5 kB | 37.5 kB | N/A |
| edge-runtime..pack.js gzip | 1.92 kB | 1.92 kB | ✓ |
| Overall change | 2.07 kB | 2.07 kB | ✓ |

</details>

<details>
<summary><strong>Next Runtimes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 01-17-Use_snapshots_for_component-stack_tests | Change |
| - | - | - | - |
| app-page-exp...dev.js gzip | 169 kB | 169 kB | ✓ |
| app-page-exp..prod.js gzip | 95.4 kB | 95.4 kB | ✓ |
| app-page-tur..prod.js gzip | 96 kB | 96 kB | ✓ |
| app-page-tur..prod.js gzip | 90.6 kB | 90.6 kB | ✓ |
| app-page.run...dev.js gzip | 142 kB | 142 kB | ✓ |
| app-page.run..prod.js gzip | 90 kB | 90 kB | ✓ |
| app-route-ex...dev.js gzip | 24.2 kB | 24.2 kB | ✓ |
| app-route-ex..prod.js gzip | 16.8 kB | 16.8 kB | ✓ |
| app-route-tu..prod.js gzip | 16.8 kB | 16.8 kB | ✓ |
| app-route-tu..prod.js gzip | 16.4 kB | 16.4 kB | ✓ |
| app-route.ru...dev.js gzip | 23.6 kB | 23.6 kB | ✓ |
| app-route.ru..prod.js gzip | 16.4 kB | 16.4 kB | ✓ |
| pages-api-tu..prod.js gzip | 9.39 kB | 9.39 kB | ✓ |
| pages-api.ru...dev.js gzip | 9.67 kB | 9.67 kB | ✓ |
| pages-api.ru..prod.js gzip | 9.39 kB | 9.39 kB | ✓ |
| pages-turbo...prod.js gzip | 22 kB | 22 kB | ✓ |
| pages.runtim...dev.js gzip | 22.6 kB | 22.6 kB | ✓ |
| pages.runtim..prod.js gzip | 22 kB | 22 kB | ✓ |
| server.runti..prod.js gzip | 49.6 kB | 49.6 kB | ✓ |
| Overall change | 942 kB | 942 kB | ✓ |

</details>

</details>
Commit: 3299d04182f0ce9e00a2e8419fb74803f4c501a5

File: test/development/acceptance-app/component-stack.test.ts (hunk 1 above)
Comment (timneutkens) [reply test/development/acceptance-app/component-stack.test.ts]: Fixed 💯 The parsing of lines assumed `:` would exist at the end only and `http://myurl/_next/etc:5:10` has `:` for the protocol 👍

**Review by sokra — APPROVED** [review]

**Review by huozhi — APPROVED** [review]

**Pull request merged**